### ISSUE TITLE
Rename dims independently from coords?

### ISSUE THREAD

**Issue by nedclimaterisk** [pydata/xarray#3026]
I have a dataset that looks like this:

```python
<xarray.Dataset>
Dimensions:  (lat: 226, lon: 261, time: 7300)
Coordinates:
  * lat      (lat) float32 -32.0 -31.9 -31.8 -31.7 ... -9.700001 -9.6 -9.5
  * lon      (lon) float32 132.0 132.1 132.2 132.3 ... 157.7 157.8 157.9 158.0
  * time     (time) object 1980-01-01 15:00:00 ... 1999-12-31 15:00:00
Data variables:
    rnd24    (time, lat, lon) float32 ...
```

#### Problem description

I would like to be able to rename the dataset dimensions, *without renaming the coordinates*.

#### Expected Output

```python
<xarray.Dataset>
Dimensions:  (y: 226, x: 261, time: 7300)
Coordinates:
  * lat      (y) float32 -32.0 -31.9 -31.8 -31.7 ... -9.700001 -9.6 -9.5
  * lon      (x) float32 132.0 132.1 132.2 132.3 ... 157.7 157.8 157.9 158.0
  * time     (time) object 1980-01-01 15:00:00 ... 1999-12-31 15:00:00
Data variables:
    rnd24    (time, y, x) float32 ...
```

As far as I can tell, there is no way to do this. I can `rename` the existing dims/coords to x/y, and then manually create new coordinates that are copies of x and y, which gets me to:

```python
<xarray.Dataset>
Dimensions:  (time: 7300, x: 261, y: 226)
Coordinates:
  * y        (y) float32 -32.0 -31.9 -31.8 -31.7 ... -9.700001 -9.6 -9.5
  * x        (x) float32 132.0 132.1 132.2 132.3 ... 157.7 157.8 157.9 158.0
  * time     (time) object 1980-01-01 15:00:00 ... 1999-12-31 15:00:00
    lat      (y) float32 -32.0 -31.9 -31.8 -31.7 ... -9.700001 -9.6 -9.5
    lon      (x) float32 132.0 132.1 132.2 132.3 ... 157.7 157.8 157.9 158.0
Data variables:
    rnd24    (time, y, x) float32 ...
```

But it doesn't seem to be possible to re-assign the new coordinates as the indexes for the existing dims.

In this case, it may seem a bit redundant, because the coordinates are equal to the grid. But I'm trying to get this output to work with code that also deals with other datasets that have non- rectilinear grids.

**Comment by shoyer** [comment]
Yes, this would be really nice. We should have a separate `rename_dims()` method that only renames dimensions.

**Comment by nedclimaterisk** [comment]
Is there even any work-around at the moment? I could try a pull request, if I could figure it out.

**Comment by jukent** [comment]
I am looking into this, but first I am making sure I understand the relationship between coordinates and dimensions. Could you give me an example of when you would use this functionality?

**Comment by jukent** [comment]
Have you tried using the xarray.dataset.swap_dims method?

**Comment by jukent** [comment]
<img width="1015" alt="Screen Shot 2019-06-20 at 3 49 00 PM" src="https://user-images.githubusercontent.com/46687291/59883664-0f604400-9373-11e9-8d7d-fbafd4437e39.png">
<img width="1007" alt="Screen Shot 2019-06-20 at 3 49 14 PM" src="https://user-images.githubusercontent.com/46687291/59883665-0ff8da80-9373-11e9-9524-747d738123a6.png">

**Comment by jukent** [comment]
Is this what you did^^ ?

**Comment by nedclimaterisk** [comment]
For example, you might have a rectilinear grid in a non-geodetic coordinate system, for example a rotated pole:

```<xarray.Dataset>
Dimensions:       (period: 7, x: 3, y: 3)
Coordinates:
    height        float64 2.0
    lon           (y, x) float64 137.6 138.1 138.6 137.6 ... 137.7 138.1 138.6
    lat           (y, x) float64 -21.71 -21.73 -21.74 ... -20.83 -20.85 -20.86
  * y             (y) float64 7.92 8.36 8.8
  * x             (x) float64 176.5 176.9 177.4
  * period        (period) object '1961-1980' '1981-2000' ... '2081-2100'
Data variables:
    loc           (period, y, x) float64 311.2 311.4 311.2 ... 316.0 316.0 315.6
    scale         (period, y, x) float64 1.231 1.223 1.192 ... 0.8923 0.8934
    shape         (period, y, x) float64 0.4358 0.4261 ... -0.001062 0.009387
    rotated_pole  |S1 b''
```

Obviously this case is a little more complicated, because x/y and lon/lat don't have a one-to-one correspondence. But I want to be able to write code that works with both cases, ideally without special casing stuff. Being able to have separately named dim and coordinate variables would help a lot.

However, now that I look at this example, I guess in my case I have both x/y and lat/lon in the coordinates anyway, so maybe I should just duplicate them, with different names.

**Comment by jukent** [comment]
Should we close this issue?

**Comment by shoyer** [comment]
I think `swap_dims` solves many of these use-cases, but it would be better to have a `rename_dims` method that worked more directly.

**Comment by jukent** [comment]
I will submit a pull request to add separate rename dims and rename coords functions in a day or two.

**Comment by jukent** [comment]
Created pull request #3042

**Comment by jukent** [comment]
This issue can be closed, and is addressed in pull request #3045

**Comment by dcherian** [comment]
Hi @jukent, if you edit the first comment of your pull request you should be able to change it to Closes #3026 which will automatically close this issue when that PR is merged.
